src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -238,17 +238,16 @@ internal bool TryRead(ref Utf8JsonReader reader, Type typeToConvert, JsonSeriali
             state.Push();
             Debug.Assert(Type == state.Current.JsonTypeInfo.Type);
 
-#if DEBUG
-            // For performance, only perform validation on internal converters on debug builds.
             if (!isContinuation)
             {
+#if DEBUG
+                // For performance, only perform token type validation of converters on debug builds.
                 Debug.Assert(state.Current.OriginalTokenType == JsonTokenType.None);
                 state.Current.OriginalTokenType = reader.TokenType;
```

Comment:
We now need to store this value in all builds to track resumable skip operations.